### PR TITLE
examples: add brotli

### LINKED ISSUE #16747: Add brotli sandbox example
### description

it would be very helpful to add a brotli sandbox

it could be fairly closely modelled on the gzip sandbox just added #16440 - although it would also require TLS i think

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

Commit Message:

Add brotli example.

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fixes #16747
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@daixiang0 ive marked this as draft as there are quite a few fairly obvious problems with it - and ci is broke

please mark as "ready for review" when you have fixed ci and done a thorough review yourself

File: docs/root/start/sandboxes/brotli.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+The sandbox is configured with two ports serving Envoy’s admin and statistics interface:
+
+- ``9901`` exposes the standard admin interface
+- ``9902`` exposes a compressed version of the admin interface
+
+Use ``curl`` to make a request for uncompressed statistics on port ``9901``, it should not contain the ``content-encoding`` header in the response:
+
+.. code-block:: console
+
+    $ curl -si -H "ccept-Encoding: br" localhost:9901/stats/prometheus | grep "content-encoding"
```
Comment (rojkov) [inline docs/root/start/sandboxes/brotli.rst:78]:
```suggestion
    $ curl -si -H "Accept-Encoding: br" localhost:9901/stats/prometheus | grep "content-encoding"
```

**Review by rojkov — COMMENTED** [review]
Nice!

Also it could be useful to mention that having two generic compressor filters in one filter chain with different compressor libraries, but otherwise identical, is a possibility too (to serve requests like `Accept-Encoding: gzip;q=0.5, br`).

File: docs/root/start/sandboxes/brotli.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    $ curl -si -H "Accept-Encoding: br" localhost:10000/file.txt | grep "content-encoding"
+
+It still works when specify mulitiple compressor filters:
+
+.. code-block:: console
+
+    $ curl -si -H "Accept-Encoding: gzip;q=0.5, br" localhost:10000/file.json | grep "content-encoding"
+    content-encoding: br
+
```
Comment (rojkov) [inline docs/root/start/sandboxes/brotli.rst:72]: Hmm... in hindsight this doesn't look like a good idea after all. The example config contains only one compression filter per filter chain and mentioning gzip in this context may confuse readers. Better drop these lines.

Requests with `Accept-Encoding: gzip;q=0.5, br` could be illustrative for setups featuring more than one compression filter in a filter chain.

Comment (phlax) [reply docs/root/start/sandboxes/brotli.rst:72]: these examples should be as simple as possible and provide an isolated setup that is exemplary of what is being shown

i think in this case we just want the `br` `accept-encoding` header

Comment (daixiang0) [reply docs/root/start/sandboxes/brotli.rst:72]: Remove them.

Comment (daixiang0) [reply docs/root/start/sandboxes/brotli.rst:72]: sorry for force push :(

**Comment by phlax** [comment]
docs are rendered here https://storage.googleapis.com/envoy-pr/16873/docs/index.html

File: docs/root/start/sandboxes/brotli.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Now, use ``curl`` to make a request for the compressed statistics:
+
+.. code-block:: console
+
+    $ curl -ski -H "Accept-Encoding: br" https://localhost:9902/stats/prometheus | grep "content-encoding"
+    content-encoding: br
+
+.. seealso::
+   :ref:`Gzip API <envoy_v3_api_msg_extensions.compression.brotli.compressor.v3.Brotli>`
```
Comment (phlax) [inline docs/root/start/sandboxes/brotli.rst:88]: i think this should be `Brotli API`

**Review by phlax — CHANGES_REQUESTED** [review]
One small nit - otherwise lgtm

Really nice work - thankyou very much @daixiang0

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Pull request merged**